test/extensions/filters/http/router/auto_sni_integration_test.cc

```diff
@@ -0,0 +1,91 @@
+                               public Event::TestUsingSimulatedTime,
+                               public HttpIntegrationTest {
+public:
+  AutoSniIntegrationTest() : HttpIntegrationTest(Http::CodecClient::Type::HTTP1, GetParam()) {}
+
+  void setup() {
+    setUpstreamProtocol(FakeHttpConnection::Type::HTTP1);
+
+    config_helper_.addConfigModifier([](envoy::config::bootstrap::v2::Bootstrap& bootstrap) {
+      auto& cluster_config = bootstrap.mutable_static_resources()->mutable_clusters()->at(0);
+      cluster_config.mutable_common_http_protocol_options()->set_auto_sni(true);
+
+      envoy::api::v2::auth::UpstreamTlsContext tls_context;
+      auto* validation_context =
+          tls_context.mutable_common_tls_context()->mutable_validation_context();
+      validation_context->mutable_trusted_ca()->set_filename(
+          TestEnvironment::runfilesPath("test/config/integration/certs/upstreamcacert.pem"));
+      cluster_config.mutable_transport_socket()->set_name("envoy.transport_sockets.tls");
+      cluster_config.mutable_transport_socket()->mutable_typed_config()->PackFrom(tls_context);
+    });
+
+    HttpIntegrationTest::initialize();
+    test_server_->waitForCounterEq("cluster_manager.cluster_added", 1);
+    test_server_->waitForGaugeEq("cluster_manager.warming_clusters", 0);
+  }
+
+  void createUpstreams() override {
+    fake_upstreams_.emplace_back(new FakeUpstream(
+        createUpstreamSslContext(), 0, FakeHttpConnection::Type::HTTP1, version_, timeSystem()));
+    fake_upstreams_[0]->set_allow_unexpected_disconnects(true);
+  }
+
+  Network::TransportSocketFactoryPtr createUpstreamSslContext() {
+    envoy::api::v2::auth::DownstreamTlsContext tls_context;
+    auto* common_tls_context = tls_context.mutable_common_tls_context();
+    auto* tls_cert = common_tls_context->add_tls_certificates();
+    tls_cert->mutable_certificate_chain()->set_filename(
+        TestEnvironment::runfilesPath("test/config/integration/certs/upstreamcert.pem"));
+    tls_cert->mutable_private_key()->set_filename(
+        TestEnvironment::runfilesPath("test/config/integration/certs/upstreamkey.pem"));
+
+    auto cfg = std::make_unique<Extensions::TransportSockets::Tls::ServerContextConfigImpl>(
+        tls_context, factory_context_);
+
+    static Stats::Scope* upstream_stats_store = new Stats::IsolatedStoreImpl();
+    return std::make_unique<Extensions::TransportSockets::Tls::ServerSslSocketFactory>(
+        std::move(cfg), context_manager_, *upstream_stats_store, std::vector<std::string>{});
+  }
+};
+
+INSTANTIATE_TEST_SUITE_P(IpVersions, AutoSniIntegrationTest,
+                         testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
+                         TestUtility::ipTestParamsToString);
+
+TEST_P(AutoSniIntegrationTest, Test) {
+  setup();
+  codec_client_ = makeHttpConnection(lookupPort("http"));
+  const auto hostname =
+      fmt::format("localhost:{}", fake_upstreams_[0]->localAddress()->ip()->port());
```

Comment:
Is SNI supposed to have port? I wonder if port is supposed to be stripped in cases like this? I'm not sure if the RFC specifies this. cc @lizan @PiotrSikora

Reply:
@mattklein123 It seems that this is invalid because SNI can support only DNS name but literal IPv4/IPv6 can't support. Port should not be included. I'll try to fix them. https://tools.ietf.org/html/rfc6066#section-3 @lizan @PiotrSikora Just to be sure, check it and waiting for your insight about this.

Reply:
No SNI shouldn't have port so we should always strip the port from `:authority`.